mod/state-transition/pkg/core/types.go

```diff
@@ -129,15 +129,20 @@ type Context interface {
 
 // Deposit is the interface for a deposit.
 type Deposit[
+	DepositT any,
 	ForkDataT any,
 	WithdrawlCredentialsT ~[32]byte,
 ] interface {
+	// Equals returns true if the Deposit is equal to the other.
+	Equals(DepositT) bool
 	// GetAmount returns the amount of the deposit.
 	GetAmount() math.Gwei
 	// GetPubkey returns the public key of the validator.
 	GetPubkey() crypto.BLSPubkey
 	// GetWithdrawalCredentials returns the withdrawal credentials.
 	GetWithdrawalCredentials() WithdrawlCredentialsT
+	// GetIndex returns deposit index
```

Comment:
needed to check sequence in genesis deposits

Reply:
also applied for use in non-genesis. Order of deposits processing, by index, should always be maintained.